File: LeetCode/python/lc_0283.py
"""
 * @file     lc_0283.py
 * @brief    
 * @author   YongDu
 * @date     2021-10-29
"""


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        index = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[index] = nums[i]
                index += 1

        for i in range(index, len(nums)):
            nums[index] = 0
            index += 1

File: LeetCode/cpp/lc_0283.cpp
/**
 * @file     lc_0283.cpp
 * @brief    https://leetcode-cn.com/problems/move-zeroes/
 * @author   YongDu
 * @date     2021-09-05
 */
class Solution {
  public:
    void moveZeroes(vector<int>& nums) {
        int index = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] != 0) {
                nums[index++] = nums[i];
            }
        }
        for (int i = index; i < nums.size(); ++i) {
            nums[i] = 0;
        }
        return;
    }
};
